### PR TITLE
added span subwords

### PR DESCRIPTION
**Pull request by SoluMilken**
Span each element in input_sequence to several subwords according to the input vocabulary.

If the element or part of element is not found in vocabulary,
the unk_token would be returned.

Requirement: All elements in input_sequence should not contain whitespaces and accent tokens.

This class is almost the same as WordPieceTokenizer in BERT.
ref: https://github.com/google-research/bert/blob/master/tokenization.py

E.g.
```python
>>> from uttut.pipeline.ops.span_subwords import SpanSubwords
>>> op = SpanSubwords(vocab={"I": 0, "apple": 1, "##s": 2}, unk='<unk>', maxlen_per_token=5)
>>> output_seq, output_labels, realigner = op.transform(["I", "like", "apples"], [1, 2, 3])
>>> output_seq
["I", "<unk>", "apple", "##s"]
>>> output_labels
[1, 2, 3, 3]
>>> realigner(output_labels)
[1, 2, 3]
```

### CODE REVIEW

File: uttut/pipeline/ops/span_subwords.py [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+        for i, token in enumerate(input_lst):
+            subtokens = span_subwords(
+                word=token,
+                unk_token=self.unk_token,
+                vocab=self.vocab,
+                max_input_chars_per_word=self.maxlen_per_token,
+            )
+            if subtokens != [token]:
+                replacement_group.add(start=i, end=i + 1, new_value=subtokens,
+                                      annotation='span-subwords')
```
Comment (GBLin5566) [inline uttut/pipeline/ops/span_subwords.py]: style here

File: uttut/pipeline/ops/span_subwords.py [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+from ..edit import lst2lst
+from ..edit.replacement import ReplacementGroup
+from ..edit.label_propagation import propagate_by_replacement_group
+
+
+class SpanSubwords(Operator):
+
+    def __init__(
+            self,
+            vocab: Dict[str, int],
```
[the remark this answers is not in the document]
Comment (SoluMilken) [reply uttut/pipeline/ops/span_subwords.py]: 後面的token2index 也要拿到

File: uttut/pipeline/ops/span_subwords.py (hunk 1 above)
Comment (SoluMilken) [reply uttut/pipeline/ops/span_subwords.py]: :ok_woman:

File: uttut/pipeline/ops/span_subwords.py [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+    using the given vocabulary.
+
+    The default arguments in BERT are
+    unk_token = '[UNK]'
+    max_input_chars_per_word = 200
+
+    E.g.
+    >>> span_subwords(
+        word="unaffable",
+        vocab={"un": 0, "##aff": 1, "##able": 2},
```
Comment (absolutelyNoWarranty) [inline uttut/pipeline/ops/span_subwords.py]: `unk_token = '[UNK]'`

File: uttut/pipeline/ops/span_subwords.py [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+    # subword matching
+    chars = list(word)
+    is_bad = False
+    start = 0
+    sub_tokens = []
+    while start < len(chars):
+        end = len(chars)
+        cur_substr = None
+        while start < end:
+            substr = "".join(chars[start: end])
```
Comment (absolutelyNoWarranty) [inline uttut/pipeline/ops/span_subwords.py]: `chars = list(word) ` and `substr = "".join(chars[start:end])`

can just be `substr = word[start:end]`

Comment (SoluMilken) [reply uttut/pipeline/ops/span_subwords.py]: This code is completely copied from BERT. Should I modify it?

File: uttut/pipeline/ops/span_subwords.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (SoluMilken) [reply uttut/pipeline/ops/span_subwords.py]: 這個 vocabulary dictionary 是 BERT 要求要給的

File: uttut/pipeline/ops/span_subwords.py (hunk 3 above)
Comment (SoluMilken) [reply uttut/pipeline/ops/span_subwords.py]: ed1b0fb

**Pull request merged**